### PR TITLE
Fixes issue #58 - Add String digits support for GenericDigits

### PR DESCRIPTION
**Pull request by gfigiel**
Hello,
I created BCD String support for GenericDigits based on the modified code of https://gist.github.com/neuro-sys/953548.

Also updated unit tests in order to test the change and cover ODD/EVEN digits encoding.

### CODE REVIEW

**Comment by deruelle** [comment]
@vetss please review

File: isup/isup-impl/src/main/java/org/mobicents/protocols/ss7/isup/util/BcdHelper.java [hunk 1]
```diff
@@ -0,0 +1,119 @@
+/**
+ *  Copyright 2016 Grzegorz Figiel
+ *
```
Comment (deruelle) [inline isup/isup-impl/src/main/java/org/mobicents/protocols/ss7/isup/util/BcdHelper.java]: this license header is wrong and can't be accepted. Please use the standard TeleStax  license header from https://docs.google.com/document/d/1RZz2nd2ivCK_rg1vKX9ansgNF6NpK_PZl81GxZ2MSnM/edit#heading=h.iayg4mgzyzs8 for any changes done to any of the files modified that don't contain this header or any new files contributed

Comment (gfigiel) [inline isup/isup-impl/src/main/java/org/mobicents/protocols/ss7/isup/util/BcdHelper.java]: Hello,
As I do not know all flavours of the OpenSource licencies I used the license header as in the code I modified https://gist.github.com/neuro-sys/953548 as not to contradict the original license (Apache 2.0). If you know that I can safly change the license into Telestax then I can change the header.

**Comment by gfigiel** [comment]
One limitation I know on my enhancement is lack of hex digits support. Should I add some note on this in the code?

File: isup/isup-impl/src/main/java/org/mobicents/protocols/ss7/isup/util/BcdHelper.java (hunk 1 above)
Comment (deruelle) [inline isup/isup-impl/src/main/java/org/mobicents/protocols/ss7/isup/util/BcdHelper.java]: Did you make any modifications to the original file ?

**Comment by deruelle** [comment]
Yes please or implementing now the hex digits support would be good.

File: isup/isup-impl/src/main/java/org/mobicents/protocols/ss7/isup/util/BcdHelper.java (hunk 1 above)
Comment (gfigiel) [inline isup/isup-impl/src/main/java/org/mobicents/protocols/ss7/isup/util/BcdHelper.java]: Yes I did modified the original. I described the modification and link to original file in Class comments.

**Comment by gfigiel** [comment]
OK I will try to make it complete (include hex digits as well) accordint to:
-- AssistingSSPIPRoutingAddress in EstablishTemporaryConnection and CorrelationID in 
-- AssistRequestInstructions may contain a Hex B digit as address signal. Refer to 
-- Annex A.6 for the usage of the Hex B digit.

File: isup/isup-impl/src/main/java/org/mobicents/protocols/ss7/isup/util/BcdHelper.java (hunk 1 above)
Comment (deruelle) [inline isup/isup-impl/src/main/java/org/mobicents/protocols/ss7/isup/util/BcdHelper.java]: Then you can change the license header. but do keep the comments on what was changed over the original Apache license file.

**Comment by deruelle** [comment]
Thanks @gfigiel

**Comment by gfigiel** [comment]
I reimplemented the BCDHelper class on my own and added appropriate IPR header.

Support for all hex digits added and telco chars # and * according to:
CAP GenerisDigits Wrapper comments:
ISUP GenericNumber & GenericDigits wrapper 
Digits {PARAMETERS-BOUND : bound} ::= OCTET STRING (SIZE( bound.&minDigitsLength .. bound.&maxDigitsLength)) SIZE: 2..16 
-- Indicates the address signalling digits. -- Refer to ETSI EN 300 356-1 [23] Generic Number & Generic Digits parameters for encoding. 
-- The coding of the subfields 'NumberQualifier' in Generic Number and 'TypeOfDigits' in 
-- Generic Digits are irrelevant to the CAP; 
-- the ASN.1 tags are sufficient to identify the parameter. 
-- The ISUP format does not allow to exclude these subfields, 
-- therefore the value is network operator specific. 
-- 
-- The following parameters shall use Generic Number: 
-- - AdditionalCallingPartyNumber for InitialDP 
-- - AssistingSSPIPRoutingAddress for EstablishTemporaryConnection 
-- - CorrelationID for AssistRequestInstructions 
-- - CalledAddressValue for all occurrences, CallingAddressValue for all occurrences. 
-- 
-- The following parameters shall use Generic Digits: 
-- - CorrelationID in EstablishTemporaryConnection 
-- - number in VariablePart 
-- - digitsResponse in ReceivedInformationArg 
-- - midCallEvents in oMidCallSpecificInfo and tMidCallSpecificInfo 
-- 
-- In the digitsResponse and midCallevents, the digits may also include the '*', '#', 
-- a, b, c and d digits by using the IA5 character encoding scheme. If the BCD even or 
-- BCD odd encoding scheme is used, then the following encoding shall be applied for the 
-- non-decimal characters: 1011 (*), 1100 (#). 
-- 
-- AssistingSSPIPRoutingAddress in EstablishTemporaryConnection and CorrelationID in 
-- AssistRequestInstructions may contain a Hex B digit as address signal. Refer to 
-- Annex A.6 for the usage of the Hex B digit. 
-- 
-- Note that when CorrelationID is transported in Generic Digits, then the digits shall 
-- always be BCD encoded.

**Comment by deruelle** [comment]
great @gfigiel . @vetss will review

**Pull request closed without merging**